realm/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -40,9 +40,9 @@
         MEM_ONLY(1);
         //ASYNC(2); // TODO: re-enable when possible
 
-        final int value;
+        public final int value;
```

Comment:
There is a lot of arguments about that particular guideline, and while it might make sense for the Android framework team, in 99.9% of other cases it wont. What that guideline is for is warning that enums have a larger memory footprint than ints, which is important if you start start creating new ones all over the place. We are nowhere near that, and enums make the code much more readable and maintainable imo.